File: 50 Coding Challenge Part I/TypeScript/Challenge14.ts
// Challenge 14: Print the first 10 Fibonacci numbers without using recursion.

let f0: number = 0;
let f1: number = 1;

console.log(f0);
console.log(f1);

for (let i = 2; i < 10; i++){

    let fi: number = f1 + f0;
    console.log(fi)

    // swaping
    f0 = f1;
    f1 = fi
    
}

File: 50 Coding Challenge Part I/dart/lib/challenge14.dart
/*  
  Coding Challenge 14: Print the first 10 Fibonacci numbers without using recursion.
  Reminder:
  F(0) = 0
  F(1) = 1
  F(n) = F(n-1) + F(n-2)

  @author: mofirojean
*/

void printFabonnaciSequence() {
  var f0 = 0;
  print(f0);
  var f1 = 1;
  print(f1);

  for (var i = 2; i <= 10; i++) {
    var fn = f0 + f1;
    print(fn);
    f0 = f1;
    f1 = fn;
  }
}
